### ISSUE TITLE
Debugging Issues: Turnstile Verification and Multi-Slide Post Downloads

### ISSUE THREAD

**Issue by 4801f421** [x404xx/Insta-Down#1]
When trying to download Instagram stories using the code, I encounter an error. Upon debugging the code, I discovered that the error is due to a failure in verifying something called 'turnstile,' which appears to be a mechanism from Cloudflare. It seems that this type of authentication is specifically used for downloading stories.

Additionally, the source code currently lacks the capability to download multi-slide posts. This issue can be easily resolved by adding lists and loops. I might edit this section and send you a pull request in the near future for inclusion in the project.

Moreover, it would be advisable to add error handling to the project, as some errors are due to internet issues and can be resolved with retries.

**Comment by x404xx** [comment]
> When trying to download Instagram stories using the code, I encounter an error. Upon debugging the code, I discovered that the error is due to a failure in verifying something called 'turnstile,' which appears to be a mechanism from Cloudflare. It seems that this type of authentication is specifically used for downloading stories.
> 
> 
> 
> Additionally, the source code currently lacks the capability to download multi-slide posts. This issue can be easily resolved by adding lists and loops. I might edit this section and send you a pull request in the near future for inclusion in the project.
> 
> 
> 
> Moreover, it would be advisable to add error handling to the project, as some errors are due to internet issues and can be resolved with retries.

Yeah, now their website has been updated. They use cloudflare turnstiles. 

![image](https://github.com/x404xx/Insta-Down/assets/114883816/eb45e064-6a1e-4b6b-a3c1-a9cbe268283b)

**Comment by 4801f421** [comment]
Thank you for the update and for letting me know about the Cloudflare turnstiles.

Regarding the issues I mentioned earlier:

The error caused by the failure to verify Cloudflare's turnstile when trying to download Instagram stories.
The current source code's lack of capability to download multi-slide posts.(would you prefer if I proceed with the necessary changes and submit a pull request for your review?)
The need for improved error handling, particularly for internet-related issues that can be resolved with retries.

Do you plan to work on resolving these issues?

**Comment by x404xx** [comment]
> Thank you for the update and for letting me know about the Cloudflare turnstiles.
> 
> Regarding the issues I mentioned earlier:
> 
> The error caused by the failure to verify Cloudflare's turnstile when trying to download Instagram stories. The current source code's lack of capability to download multi-slide posts.(would you prefer if I proceed with the necessary changes and submit a pull request for your review?) The need for improved error handling, particularly for internet-related issues that can be resolved with retries.
> 
> Do you plan to work on resolving these issues?

Yes, you can. Go ahead; I will review it.

By the way, I already made a turnstile bypasser API; check the link below.

https://github.com/x404xx/Turnstile-Solver
